test/js-native-api/test_typedarray/test_typedarray.c

```diff
@@ -230,15 +230,58 @@ static napi_value IsDetached(napi_env env, napi_callback_info info) {
   return result;
 }
 
+static napi_value BufferFromArrayBuffer(napi_env env, napi_callback_info info) {
+  napi_status status;
+  napi_value arraybuffer;
+  void* data;
+  napi_value buffer;
+  size_t byte_length = 1024;
+  size_t byte_offset = 0;
+
+  status = napi_create_arraybuffer(env, byte_length, &data, &arraybuffer);
+  NODE_API_ASSERT(env, status == napi_ok, "Failed to create arraybuffer");
+
+  status = napi_create_buffer_from_arraybuffer(
+      env, byte_length, arraybuffer, byte_offset, &buffer);
+  NODE_API_ASSERT(
+      env, status == napi_ok, "Failed to create buffer from arraybuffer");
+
+  void* buffer_data;
+  size_t buffer_length;
+  status = napi_get_buffer_info(env, buffer, &buffer_data, &buffer_length);
+  NODE_API_ASSERT(env, status == napi_ok, "Failed to get buffer info");
+  NODE_API_ASSERT(env, buffer_length == byte_length, "Buffer length mismatch");
+
+  status = napi_create_buffer_from_arraybuffer(
+      env, byte_length, arraybuffer, byte_length + 1, &buffer);
+  NODE_API_ASSERT(env,
+                  status == napi_range_error,
+                  "Expected range error for invalid byte offset");
+
+  napi_value non_arraybuffer;
+  status = napi_create_uint32(env, 123, &non_arraybuffer);
+  NODE_API_ASSERT(
+      env, status == napi_ok, "Failed to create non-arraybuffer value");
+
+  status = napi_create_buffer_from_arraybuffer(
+      env, byte_length, non_arraybuffer, byte_offset, &buffer);
+  NODE_API_ASSERT(env,
+                  status == napi_invalid_arg,
+                  "Expected invalid arg error for non-arraybuffer input");
```

Comment:
chech that `buffer` is a `Buffer` with `napi_is_buffer`

Reply:
I checked as you said